test/integration/daemonset/daemonset_test.go

```diff
@@ -393,3 +397,267 @@ func TestInsufficientCapacityNodeDaemonDoesNotLaunchPod(t *testing.T) {
 		close(stopCh)
 	}
 }
+
+func getPods(t *testing.T, podClient corev1typed.PodInterface, labelMap map[string]string) *v1.PodList {
```

Comment:
Should we put all helper functions in a new file `test/integration/daemonset/util.go` like deployment does, or in a more generic new test util file `test/utils/daemonset.go`? I prefer latter because there are some functions that we definitely can share among both e2e and integration tests. However, it seems that latter requires non-tedious code refactoring to existing daemonset e2e tests (it may be wise to do the refactoring in another PR).

Reply:
I had https://github.com/kubernetes/kubernetes/pull/55089/files which put all the workload controllers into an `apps/` directory and had `apps/util.go` for these methods. Let me see if I can revive that.

Reply:
Great, that would be helpful!